✨ Add DatadogClient for tracking http.Client calls (#262)
DataDog/dd-sdk-flutter: packages/datadog_tracking_http_client/README.md

)..enableHttpTracking()
```

## Using http.Client wrapping

This package also supplies a composable client usable with the [http pub package](https://pub.dev/packages/http) called `DatadogClient`. For most scenarios, Datadog recommends you use the HTTP tracking method above, but there are a few scenarios where using `DatadogClient` might make more sense:

> * If you are using native HTTP libraries like [cronet_http](https://pub.dev/packages/cronet_http) or [cupertino_http](https://pub.dev/packages/cupertino_http), which do not work with the above tracking method 

Reviewer: ```suggestion
* If you are using native HTTP libraries like [cronet_http](https://pub.dev/packages/cronet_http) or [cupertino_http](https://pub.dev/packages/cupertino_http), which do not work with the above tracking method. 
```

> * It you only want to track specific resource requests.

Reviewer 2: ```suggestion
* If you only want to track specific resource requests.
```

If you are using `cronet_http` or `cupertino_http`, you can combine `DatadogClient` with the above tracking method. Otherwise, the two methods may interfere with each other.

To use the `DatadogClient`, create and compose the `Client` from the `http` package:

```dart
import 'package:datadog_tracking_http_client/datadog_tracking_http_client.dart';
import 'package:http/http.dart' as http;

final configuration = DdSdkConfiguration(
  // specifying firstPartyHosts is still necessary to
  // enable distributed tracing
  firstPartyHosts: ['example.com'],
)

final httpClient = http.Client()
final datadogClient = DatadogClient(datadogSdk: DatadogSdk.instance, innerClient: httpClient);
```

> The `innerClient` parameter is optional. If it is not supplied, DatadogClient will create a `Client` for you with default options.  Now, you can use `datadogClient` to make any network calls, and they will be tracked.

Reviewer 2: ```suggestion
The `innerClient` parameter is optional. If it is not supplied, `DatadogClient` creates a `Client` for you with default options.  After configuring the `Client`, you can use `DatadogClient` to make any network calls, and they will be tracked.
```
